### ISSUE TITLE
It is apparently impossible to create a route in the app router with an at (@) character in it

### ISSUE THREAD

**Issue by joulev** [vercel/next.js#52391]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.5.0: Thu Jun  8 22:21:34 PDT 2023; root:xnu-8796.121.3~7/RELEASE_ARM64_T8112
    Binaries:
      Node: 20.3.1
      npm: 9.6.7
      Yarn: 1.22.19
      pnpm: 8.6.5
    Relevant Packages:
      next: 13.4.10-canary.0
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.1.6
    Next.js Config:
      output: N/A
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App Router

### Link to the code that reproduces this issue or a replay of the bug

N/A

### To Reproduce

Couldn't find a similar issue, but I feel like this problem should've already been reported somewhere. If you find an issue of which this issue is a duplicate, do tell.

I want to add `/@test` to my website. But `@` is already taken by parallel routing, so I need to find a way to escape it. Attempts (all are in dev mode):

1. `app/@test/page.tsx` → not working, for obvious reasons

2. `app/%40test/page.tsx`

    I tried to escape `@` using URL encoding (similar to the `_` case), but this doesn't work either. `/%40test` does work, but `/@test` doesn't.

3. `app/[dynamic]/page.tsx` with `generateStaticParams` to `{ dynamic: "@test" }` and `dynamicParams = false`

    This works, but why do I need to do so much just to get `/@test`?

4. `app/some/route/page.tsx` then rewrite `/@test` to it

    This works, but again, this is too much work.

[Related question in the Next.js Discord server](https://nextjs-forum.com/post/1126818180279578714).

### Describe the Bug

Too much effort is needed for something as trivial as adding a route with `@` in the path.

### Expected Behavior

(2) above should work.

### How are you deploying your application? (if relevant)

In this issue, only `next dev`. I didn't try prod mode.

**Comment by balazsorban44** [comment]
Thanks, `@` is a reserved symbol in App Router for [Parallel Routes](https://nextjs.org/docs/app/building-your-application/routing/parallel-routes), and the correct solution is to define the path with the encoded `%40` version.

We are missing documentation on this behavior - similar to `_`, which is also reserved - so I forwarded this to our docs team.

**Comment by balazsorban44** [comment]
I think we should add section mentioning these under https://nextjs.org/docs/app/api-reference/file-conventions

**Comment by joulev** [comment]
@balazsorban44 Yes I agree with that too, but actually `%40` doesn't actually work.

* [Reproduction repository](https://github.com/joulev/debug/tree/nextjs-escaping-at-not-working)
* Deployment: [`/@test`](https://debug-git-nextjs-escaping-at-not-working-joulev.vercel.app/@test) doesn't work but [`/%40test`](https://debug-git-nextjs-escaping-at-not-working-joulev.vercel.app/%40test) does.

**Comment by musjj** [comment]
Using `%40` still doesn't work. Is this solved in v15?

**Comment by aficus11** [comment]
Crazy. Who was that genius who reserved `@` for parallel routes without any escape hatch??

You broke most common URL convention on the web. The only way to have `/@username` or `/@/username` is middleware???
